.github/workflows/rollback.yml

```diff
@@ -1,11 +1,287 @@
+            echo "Using Terraform Workspace: novu-prod-eu"
+            echo "TF_WORKSPACE=novu-prod-eu" >> $GITHUB_ENV
+          elif [[ "${{ matrix.region }}" == "US" && "${{ github.event.inputs.environment }}" == "Production" ]]; then
+            echo "Using Terraform Workspace: novu-prod"
+            echo "TF_WORKSPACE=novu-prod" >> $GITHUB_ENV
+          elif [[ "${{ matrix.region }}" == "EU" && "${{ github.event.inputs.environment }}" == "Development" ]]; then
+            echo "Using Terraform Workspace: novu-dev"
+            echo "TF_WORKSPACE=novu-dev" >> $GITHUB_ENV
+          elif [[ "${{ matrix.region }}" == "US" && "${{ github.event.inputs.environment }}" == "Development" ]]; then
+            echo "Using Terraform Workspace: novu-dev"
+            echo "TF_WORKSPACE=novu-dev" >> $GITHUB_ENV
+            echo "Error: Development environment doesn't exist in the US region." >&2
+            exit 1
+          else
+            echo "Using Terraform Workspace: novu-dev"
+            echo "TF_WORKSPACE=novu-dev" >> $GITHUB_ENV
+          fi
+
+      - name: Checkout cloud infra
+        uses: actions/checkout@master
+        with:
+          repository: novuhq/cloud-infra
+          token: ${{ secrets.GH_PACKAGES }}
+          path: cloud-infra
+
+      - name: Terraform setup
+        uses: hashicorp/setup-terraform@v1
+        with:
+          cli_config_credentials_token: ${{ secrets.TF_API_TOKEN }}
+          terraform_version: 1.5.5
+          terraform_wrapper: false
+
+      - name: Terraform Init
+        working-directory: cloud-infra/terraform/novu/aws
+        run: terraform init
+
+      - name: Terraform get output
+        working-directory: cloud-infra/terraform/novu/aws
+        id: terraform
+        env:
+          SERVICE_NAME: ${{ github.event.inputs.service }}
+        run: |
+          echo "ecs_container_name=$(terraform output -json ${{ env.SERVICE_NAME }}_ecs_container_name | jq -r .)" >> $GITHUB_ENV
+          echo "ecs_service=$(terraform output -json ${{ env.SERVICE_NAME }}_ecs_service | jq -r .)" >> $GITHUB_ENV
+          echo "ecs_cluster=$(terraform output -json ${{ env.SERVICE_NAME }}_ecs_cluster | jq -r .)" >> $GITHUB_ENV
+          echo "task_name=$(terraform output -json ${{ env.SERVICE_NAME }}_task_name | jq -r .)" >> $GITHUB_ENV
+          echo "aws_region=$(terraform output -json aws_region | jq -r .)" >> $GITHUB_ENV
+
+      - name: Configure AWS credentials
+        uses: aws-actions/configure-aws-credentials@v4
+        with:
+          aws-access-key-id: ${{ secrets.AWS_ACCESS_KEY_ID }}
+          aws-secret-access-key: ${{ secrets.AWS_SECRET_ACCESS_KEY }}
+          aws-region: ${{ env.aws_region }}
+
+      - name: ECS get output
+        if: contains(fromJson('["Previous Version"]'), github.event.inputs.mode)
+        id: ecs-output
+        run: |
+          echo "Retrieving current_task_definition_arn..."
```

Comment:
@AliaksandrRyzhou How much time does it take to rollback? There are several API calls and we need to make sure this action is fast.

Reply:
@SokratisVidros Regarding ECS, the rollback initialization starts already in 30 seconds but other part of rollback depends on ECS update mechanism (Task definition update, health checks, Load Balancer checks etc). I've added option aws ecs wait services-stable so that we be assure that our service was deployed and full rollback takes about 4 minutes if the service is working. (https://github.com/novuhq/novu/actions/runs/8877861439/job/24372292994)
Netlify works much more faster, only a few second to make a previous deploy published. https://github.com/novuhq/novu/actions/runs/8923431167/job/24507673237